hbt/production/res_networks.py

```diff
@@ -693,10 +658,318 @@ def call_func(self, events: ak.Array, **kwargs) -> ak.Array:
+        ]
+
+        # update produced columns
+        self.produces |= set(self.output_columns)
+
+    def update_events(self, events: ak.Array) -> ak.Array:
+        events = super().update_events(events)
+
+        # ensure coffea behavior for VBFJets
+        events = self[attach_coffea_behavior](
+            events,
+            collections={"VBFJet": default_coffea_collections["Jet"]},
+        )
+
+        padded_vbf_jets = ak.pad_none(events.VBFJet, 2, axis=1)
+        events = set_ak_column(events, "padded_vbf_jets", padded_vbf_jets)
+
+        return events
+
+    def define_categorical_inputs(self, events: ak.Array, cat: DotDict) -> None:
+        super().define_categorical_inputs(events, cat)
+
+        # dm1 for e/mu is changed from -1 to -999
+        cat.dm1[cat.dm1 == -1] = -999
+
+        # add vbf jet pair presence
+        cat.has_vbf_jets = ak.num(events.VBFJet) >= 2
+
+    def define_continuous_inputs(self, events: ak.Array, cont: DotDict, cat: DotDict) -> None:
+        super().define_continuous_inputs(events, cont, cat)
+
+        # store original features as a shallow copy for reference below
+        cont_orig = cont.copy()
+
+        # remove leading met features
+        for name in ["met_px", "met_py", "met_cov00", "met_cov01", "met_cov11"]:
+            del cont[name]
+
+        # the now leading features are preserved up to bjet2_hhbtag, drop features after that
+        idx = list(cont.keys()).index("bjet2_hhbtag")
+        for name in list(cont.keys())[idx + 1:]:
+            del cont[name]
+
+        # add regressed neutrino momenta
+        cont.nu1_px = events.reg_dnn_moe_nu1_px
+        cont.nu1_py = events.reg_dnn_moe_nu1_py
+        cont.nu1_pz = events.reg_dnn_moe_nu1_pz
+        cont.nu2_px = events.reg_dnn_moe_nu2_px
+        cont.nu2_py = events.reg_dnn_moe_nu2_py
+        cont.nu2_pz = events.reg_dnn_moe_nu2_pz
+
+        # compute nu energies for later use
+        nu1_e = (cont.nu1_px**2 + cont.nu1_py**2 + cont.nu1_pz**2)**0.5
+        nu2_e = (cont.nu2_px**2 + cont.nu2_py**2 + cont.nu2_pz**2)**0.5
+
+        # fatjet features (flipped order w.r.t. original)
+        cont.fatjet_px = cont_orig.fatjet_px
+        cont.fatjet_py = cont_orig.fatjet_py
+        cont.fatjet_pz = cont_orig.fatjet_pz
+        cont.fatjet_e = cont_orig.fatjet_e
+
+        # htt features (with nu's, flipped order w.r.t. original)
+        cont.htt_regr_px = cont_orig.htt_px + cont.nu1_px + cont.nu2_px
+        cont.htt_regr_py = cont_orig.htt_py + cont.nu1_py + cont.nu2_py
+        cont.htt_regr_pz = cont_orig.htt_pz + cont.nu1_pz + cont.nu2_pz
+        cont.htt_regr_e = cont_orig.htt_e + nu1_e + nu2_e
+
+        # hbb features (flipped order w.r.t. original)
+        cont.hbb_px = cont_orig.hbb_px
+        cont.hbb_py = cont_orig.hbb_py
+        cont.hbb_pz = cont_orig.hbb_pz
+        cont.hbb_e = cont_orig.hbb_e
+
+        # htthbb features (with nu's, also note the non-flipped order)  # TODO: really?
```

Comment:
```suggestion
        # htthbb features (with nu's, also note the non-flipped order) 
```
yes, see https://gitlab.cern.ch/cclubbtautau/AnalysisCore/-/blob/19cc9e8c02dda68a37c7d9ee308100a9e1821648/src/HHRun3DNNInterface.cc#L334, but Filip was also bothered by that and sounded like he wanted to change that in the future...